magicctapipe/io/tests/test_io.py

```diff
@@ -19,619 +21,721 @@
-    for file in dl2_mc:
-        data, _, _ = load_mc_dl2_data_file(
-            str(file), "gammaness>0.1", "software", "simple"
+        for file in stereo_mc:
+            event_data = pd.read_hdf(str(file), key="events/parameters")
+            event_data.set_index(["obs_id", "event_id", "tel_id"], inplace=True)
+            event_data.sort_index(inplace=True)
+            data = get_stereo_events(event_data, config_gen)
+            assert np.all(data["multiplicity"] > 1)
+            assert np.all(data["combo_type"] >= 0)
+
+    def test_get_stereo_events_mc_cut(self, gamma_stereo, p_stereo, config_gen):
+        """
+        Check on quality cuts
+        """
+        stereo_mc = (
+            [p for p in gamma_stereo[0].glob("*")]
+            + [p for p in gamma_stereo[1].glob("*")]
+            + [p for p in p_stereo[0].glob("*")]
+            + [p for p in p_stereo[1].glob("*")]
         )
-        assert np.all(data["gammaness"] > 0.1)
-
-
-def test_load_mc_dl2_data_file_opt(p_dl2, gamma_dl2):
-    """
-    Check on event_type
-    """
-    dl2_mc = [p for p in gamma_dl2.glob("*")] + [p for p in p_dl2.glob("*")]
-    for file in dl2_mc:
-        data_s, _, _ = load_mc_dl2_data_file(str(file), "width>0", "software", "simple")
-        assert np.all(data_s["combo_type"] < 3)
+        for file in stereo_mc:
+            event_data = pd.read_hdf(str(file), key="events/parameters")
+            event_data.set_index(["obs_id", "event_id", "tel_id"], inplace=True)
+            event_data.sort_index(inplace=True)
+            data = get_stereo_events(event_data, config_gen, "intensity>50")
+            assert np.all(data["intensity"] > 50)
+
+    def test_load_train_data_files_p(self, p_stereo):
+        """
+        Check dictionary
+        """
+
+        events = load_train_data_files(str(p_stereo[0]))
+        assert list(events.keys()) == ["M1_M2", "LST1_M1", "LST1_M2", "LST1_M1_M2"]
+        data = events["LST1_M1"]
+        assert np.all(data["combo_type"] == 1)
+        assert "off_axis" in data.columns
+        assert "true_event_class" not in data.columns
+
+    def test_load_train_data_files_g(self, gamma_stereo):
+        """
+        Check dictionary
+        """
+
+        events = load_train_data_files(str(gamma_stereo[0]))
+        assert list(events.keys()) == ["M1_M2", "LST1_M1", "LST1_M2", "LST1_M1_M2"]
+        data = events["LST1_M1"]
+        assert np.all(data["combo_type"] == 1)
+        assert "off_axis" in data.columns
+        assert "true_event_class" not in data.columns
+
+    def test_load_train_data_files_off(self, gamma_stereo):
+        """
+        Check off-axis cut
+        """
+        events = load_train_data_files(
+            str(gamma_stereo[0]), offaxis_min="0.2 deg", offaxis_max="0.5 deg"
+        )
+        data = events["LST1_M1"]
+        assert np.all(data["off_axis"] >= 0.2)
+        assert np.all(data["off_axis"] <= 0.5)
```

Comment:
in the various "cut" tests I think it would be good to check also that the returned list is not empty.

Reply:
Done